File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    a = [1,2,3,4,5] * 2000
    b = [2,1,2,3,2,4,2,5] * 1250
    c = [3,3,1,1,2,2,4,4,5,5] * 1000
    d = []
    e = []
    f = []
    for i in range(len(answers)):
        if a[i] == answers[i]:
          d.append("o")
        if b[i] == answers[i]:
          e.append("o")
        if c[i] == answers[i]:
          f.append("o")
    g = max(len(d),len(e),len(f))
    if len(d) == g:
        answer.append(1)
    if len(e) == g:
        answer.append(2)
    if len(f) == g:
        answer.append(3)
    return answer

File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] a = {1,2,3,4,5};
        int[] b = {2,1,2,3,2,4,2,5};
        int[] c = {3,3,1,1,2,2,4,4,5,5};
        List<Integer> d = new ArrayList<>();
        List<Integer> e = new ArrayList<>();
        List<Integer> f = new ArrayList<>();
        for (int i = 0; i < answers.length; i++) {
            if (a[i % a.length] == answers[i]) {
                d.add(i);
            }
            if (b[i % b.length] == answers[i]) {
                e.add(i);
            }
            if (c[i % c.length] == answers[i]) {
                f.add(i);
            }
        }
        
        int g = Math.max(Math.max(d.size(), e.size()), f.size());
        ArrayList<Integer> li = new ArrayList<>();
        if (d.size() == g) {
            li.add(1);
        }
        if (e.size() == g) {
            li.add(2);
        }
        if (f.size() == g) {
            li.add(3);
        }
        
        return li.stream().mapToInt(i -> i).toArray();
    }
}
